src/usr/local/www/firewall_rules_edit.php

```diff
@@ -572,7 +572,7 @@ function is_aoadv_used($rule_config) {
 	}
 	if ($_POST['os']) {
 		if ($_POST['proto'] != "tcp") {
-			$input_errors[] = gettext("OS detection is only valid with protocol TCP.");
+			$input_errors[] = gettext("OS detection is only valid with the TCP protocol.");
```

Comment:
The "P" in TCP is "protocol", this wording reads more redundantly. Maybe cut it out and say "... only valid for TCP".